_posts/2024-02-01-wildfly-glow.adoc

```diff
@@ -0,0 +1,504 @@
+----
+
+== Datasources support, setting a datasource add-on
+
+We are using the `postgresql` add-on to fix the reported error.
+
+[source,bash]
+----
+wildfly-glow scan todo-backend.war --add-ons=postgresql
+
+Wildfly Glow is scanning...
+context: bare-metal
+enabled profile: none
+galleon discovery
+- feature-packs
+   org.wildfly:wildfly-galleon-pack:30.0.1.Final
+   org.wildfly:wildfly-datasources-galleon-pack:6.0.0.Final
+- layers
+   ee-core-profile-server
+   ejb-lite
+   jpa
+   jaxrs
+   postgresql-datasource
+
+enabled add-ons
+- postgresql : Documentation in https://github.com/wildfly-extras/wildfly-datasources-galleon-pack
+
+strongly suggested configuration
+
+postgresql-datasource:
+ - POSTGRESQL_DATABASE=Defines the database name to be used in the datasource’s `connection-url` property.
+ - POSTGRESQL_JNDI=java:jboss/datasources/ToDos
+ - POSTGRESQL_PASSWORD=Defines the password for the datasource.
+ - POSTGRESQL_USER=Defines the username for the datasource.
+
+----
+
+WildFly Glow detects that the postgresql datasource has to be configured with env variables and advertise the set of environment variables to be used when starting the server.
+
+Example of a server started with env variables set:
+
+[source,bash]
+----
+POSTGRESQL_DATABASE=test \
+POSTGRESQL_JNDI=java:jboss/datasources/ToDos \
+POSTGRESQL_PASSWORD=test \
+POSTGRESQL_USER=test \
+sh ./bin/target/server/bin/standalone.sh 
+----
+
+== Experimenting with WildFly Glow
+
+=== Using WildFly Glow CLI
+
+To get started with WildFly Glow CLI, you can download the https://github.com/wildfly/wildfly-glow/releases[latest WildFly Glow CLI]. 
+
+To scan a deployment you plan to deploy to WildFly, call:
+
+[source,bash]
+----
+wildfly-glow.sh scan <your deployment>
+----
+
+You can report issues/suggestion as https://github.com/wildfly/wildfly-glow/issues[Issues] in the WildFly Glow project.
+
+=== Using the WildFly Maven Plugin
+
+The WildFly quickstarts have been ported to rely on WildFly Glow, you should clone the following quickstart https://github.com/wildfly/quickstart/tree/port-to-glow[branch] and play with the quickstarts of your choice.
+For example:
+
+[source,bash]
+----
+cd microprofile-config
+mvn clean package -Pprovisionind-server
+----
+
+To produce a WildFly Bootable JAR call:
+
+[source,bash]
+----
+cd microprofile-config
+mvn clean package -Pbootable-jar
+----
+
+== Status
+
+For WildFly 31, WildFly Glow is at a Beta level of quality. Some facts:
+
+** https://github.com/wildfly/wildfly-glow/releases[Latest WildFly Glow] version is currently 1.0.0.Beta5.
+** https://github.com/wildfly/wildfly-maven-plugin[WildFly Maven Plugin] 5.0.0.Beta2 supports WildFly Glow and the ability to produce WildFly Bootable JAR.
+** All WildFly quickstarts have been ported to use WildFly Glow in this preview https://github.com/wildfly/quickstart/tree/port-to-glow[branch].
+** WildFly 31 testsuite has been ported to use WildFly Glow in 100% of its cases.
```

Comment:
I think we are still using Galleon provisioning in some cases. So perhaps instead of 100%:

The WildFly 31 testsuite has been ported to use WildFly Glow everywhere where it is practical.